plugins/out_s3/s3.c

```diff
@@ -1060,8 +1062,12 @@ static int upload_data(struct flb_s3 *ctx, struct s3_file *chunk,
     if (chunk) {
         create_time = chunk->create_time;
     }
-    else {
+    else if (tms == NULL) {
         create_time = time(NULL);
+        flb_plg_error(ctx->ins, "Failed to get upload time from timestamp, "
+                      "assign the current time instead.");
```

Comment:
Let's make this more clear to the user what the impact will be: note that this is the timestamp used for the S3 key

also nit: for user facing message, I'd not use the word "assign" here, that feels like "assign a value in code" which is not what we mean to the user, we just mean the timestamp we use in the S3 key.

Reply:
make sense.